cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_config.go

```diff
@@ -9,7 +9,8 @@ import (
 
 // MetricConfig provides common config for a particular metric.
 type MetricConfig struct {
-	Enabled bool `mapstructure:"enabled"`
+	Enabled bool   `mapstructure:"enabled"`
+	Name    string `mapstructure:"-"`
```

Comment:
Even if this is not available to the end user, it's still part of `Config` Go public API which is confusing. Why don't generate a separately, as part of `MetricsBuilder` maybe, e.g. `MetricsBuilder.Info.MyMetric.Name`?

Reply:
updated